probe-rs/src/core/mod.rs

```diff
@@ -117,14 +119,75 @@ pub(crate) enum RegisterKind {
 /// Creating a new `RegisterValue` should be done using From or Into.
 /// Converting a value back to a primitive type can be done with either
 /// a match arm or TryInto
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy)]
 pub enum RegisterValue {
     /// 32-bit unsigned integer
     U32(u32),
     /// 64-bit unsigned integer
     U64(u64),
 }
 
+impl RegisterValue {
+    /// A helper function to determine if the contained register value is equal to the maximum value that can be stored in that datatype.
+    pub fn is_max_value(&self) -> bool {
+        match self {
+            RegisterValue::U32(register_value) => *register_value == u32::MAX,
+            RegisterValue::U64(register_value) => *register_value == u64::MAX,
+        }
+    }
+
+    /// A helper function to determine if the contained register value is zero.
+    pub fn is_zero(&self) -> bool {
+        match self {
+            RegisterValue::U32(register_value) => register_value.is_zero(),
+            RegisterValue::U64(register_value) => register_value.is_zero(),
+        }
+    }
+}
+
+impl Default for RegisterValue {
+    fn default() -> Self {
+        // Smallest data storage as default.
+        RegisterValue::U32(0_u32)
+    }
+}
+
+impl PartialOrd for RegisterValue {
```

Comment:
Why implement it this way versus doing the comparison on extracted values?  I don't have see any issues with this, but it could get complicated if we start to support registers that have complex access patterns, like SIMD.

Reply:
To get an extracted value, it is necessary to use a `try_into()?`, and that in turn cannot be used in closures. It doesn't get used that much, so I am open to finding a different solution. Any suggestions?

Reply:
I thought about this some more and don't have a better alternative off the top of my head.  Lets go with what you have and if we need to rework this later we can do so.